src/Libraries/PythonNodeModels/PythonEvaluatorsHelper.cs

```diff
@@ -0,0 +1,100 @@
+﻿using System;
+using System.Linq;
+
+namespace PythonNodeModels
+{
+    /// <summary>
+    /// Enum of possible values of python engine versions.
+    /// TODO: Remove when dynamic loading logic is there then we no longer need a hard copy of the options
+    /// </summary>
+    public enum PythonEngineVersion
+    {
+        IronPython2,
+        CPython3
+    }
+
+    /// <summary>
+    /// Singleton class that other class can access and use for query loaded Python Engine info.
+    /// TODO: Make the Singleton class public when Dynamic loading is also ready.
+    /// </summary>
+    internal sealed class PythonEvaluatorsHelper
+    {
+        /// <summary>
+        /// Use Lazy<![CDATA[PythonEvaluatorsHelper]]> to make sure the Singleton class is only initialized once
+        /// </summary>
+        private static readonly Lazy<PythonEvaluatorsHelper>
+            lazy =
+            new Lazy<PythonEvaluatorsHelper>
+            (() => new PythonEvaluatorsHelper());
+
+        /// <summary>
+        /// The actual instance stored in the Singleton class
+        /// </summary>
+        internal static PythonEvaluatorsHelper Instance { get { return lazy.Value; } }
+
+        // TODO: Improve the following field with dynamic loading and interface
+        internal static bool IsIronPythonEnabled = false;
+        internal static string IronPythonEvaluatorClass = "IronPythonEvaluator";
+        internal static string IronPythonEvaluationMethod = "EvaluateIronPythonScript";
+
+        internal static bool IsCPythonEnabled = false;
+        internal static string CPythonEvaluatorClass = "CPythonEvaluator";
+        internal static string CPythonEvaluationMethod = "EvaluatePythonScript";
+
+        /// <summary>
+        /// Singleton class initialization logic which will be run in a lazy way the first time Dynamo try to evaluate a Python node
+        /// </summary>
+        private PythonEvaluatorsHelper()
+        {
+            var assems = AppDomain.CurrentDomain.GetAssemblies();
+            // Currently we are using try-catch to validate loaded assembly and evaluation method exist
+            // but we can optimize by checking all loaded types against evaluators interface later
+            try
+            {
+                var IronPythonAssem = assems.First(x => x.FullName.Contains("DSIronPython"));
```

Comment:
while this is dynamic in the sense that it checks if the assemblies are present without needing to reference them - it is not dynamic in all aspects. For example, we need to know the names of the assemblies to check, and only check for 2 engine versions. It would be nice if this handled the case of new engines we have not written yet - ie IronPython 3.

Is that coming in another PR or you don't think it's useful enough?

Reply:
Yes, @mjkkirschner Please refer to the TODO and comments.. We have a dedicate Jira task about Dynamic loading, this task is to make sure API is available so I left comments at places where we could apply dynamic loading later

Reply:
got it thanks.